packages/http-client-csharp/generator/Microsoft.Generator.CSharp.ClientModel/src/Providers/MrwSerializationTypeDefinition.cs

```diff
@@ -1033,22 +1033,25 @@ private MethodBodyStatement[] CreateWritePropertiesStatements()
             for (var i = 0; i < propertyCount; i++)
             {
                 var property = _model.Properties[i];
-                var propertyWireInfo = property.WireInfo;
-                var propertySerializationName = propertyWireInfo?.SerializedName ?? property.Name;
-                var propertyMember = new MemberExpression(null, propertySerializationName);
-                var propertySerializationFormat = propertyWireInfo?.SerializationFormat ?? SerializationFormat.Default;
-                var propertyIsReadOnly = propertyWireInfo?.IsReadOnly ?? false;
-                var propertyIsRequired = propertyWireInfo?.IsRequired ?? false;
+                // we should only write those properties with a wire info. Those properties without wireinfo indicate they are not spec properties.
+                if (property.WireInfo is not { } wireInfo)
+                {
+                    continue;
+                }
+                var propertySerializationName = wireInfo.SerializedName;
+                var propertySerializationFormat = wireInfo.SerializationFormat;
+                var propertyIsReadOnly = wireInfo.IsReadOnly;
+                var propertyIsRequired = wireInfo.IsRequired;
 
                 // Generate the serialization statements for the property
                 var writePropertySerializationStatements = new MethodBodyStatement[]
                 {
-                    _utf8JsonWriterSnippet.WritePropertyName(propertySerializationName.ToVariableName()),
```

Comment:
we should not change the property serialized name, otherwise we are writing a wrong JSON.
There is a case added in the test project that could break this implementation - what if the wire name of the property starts with an upper case letter?